### ISSUE TITLE
Can you share the CLIP score calculation script?

### ISSUE THREAD

**Issue by mingtan2** [mlfoundations/datacomp#20]
Would it be possible to share the CLIP score calculation script?

I tried several different settings but cannot reproduce the CLIP similarity numbers stored in the metadata. Below is the script I used for reproduction.  Thanks!

```
  import torch
  import webdataset as wds
  import pandas as pd
  from training.data import (log_and_continue,
                             get_dataset_size,
                             tarfile_to_samples_nothrow,
                             filter_no_caption_or_no_image)
  from open_clip.factory import create_model_and_transforms, get_tokenizer
  
  
  shard_root = "/path/to/medium/shards/"
  meta_root = "/path/to/medium/metadata/"
  model_name = "ViT-B-32"
  pretrained = "openai"
  precision = "fp32"
  device = "cuda:0"
  batch_size = 64
  workers = 4
  input_shards = shard_root + "/{00000000..00000000}.tar"
  
  model, _, preprocess_val = create_model_and_transforms(
      model_name,
      pretrained,
      precision=precision,
      device=device,
      jit=True,
      output_dict=True
  )
  tokenizer = get_tokenizer(model_name)
  
  num_samples, num_shards = get_dataset_size(input_shards)
  print("# shards:", num_shards)
  
  pipeline = [wds.SimpleShardList(input_shards)]
  pipeline.extend([tarfile_to_samples_nothrow])
  pipeline.extend([
      wds.select(filter_no_caption_or_no_image),
      wds.decode("pilrgb", handler=log_and_continue),
      wds.rename(image="jpg;png;jpeg;webp", text="txt", json="json"),
      wds.map_dict(image=preprocess_val, text=lambda text: tokenizer(text)[0], json=lambda data: {"uid": data["uid"]}),
      wds.to_tuple("image", "text", "json"),
      wds.batched(batch_size, partial=True)
  ])
  dataset = wds.DataPipeline(*pipeline)
  dataloader = wds.WebLoader(
      dataset,
      batch_size=None,
      shuffle=False,
      num_workers=workers,
      persistent_workers=workers > 0,
  )
  for img, txt, info in dataloader:
      with torch.no_grad():
          img = img.to(device)
          txt = txt.to(device)
          img_f = model.encode_image(img)
          txt_f = model.encode_text(txt)
  
          img_f = img_f / img_f.norm(dim=-1, keepdim=True)
          txt_f = txt_f / txt_f.norm(dim=-1, keepdim=True)
  
          sim = (img_f * txt_f).sum(-1).cpu().numpy()
  
          uid = [_["uid"] for _ in info]
  
          score_dict = {u: s for u, s in zip(uid, sim)}
  
          meta_info = pd.read_parquet(meta_root + "/0020f0cbd157d470aa56bea08e304b90.parquet", engine='pyarrow')
          for k in score_dict:
              print(score_dict[k], meta_info[meta_info["uid"] == k]['clip_b32_similarity_score'].tolist()[0])
          # examples: 
          # 0.21029426 0.2060546875
          # 0.30629587 0.29931640625
          # 0.26773185 0.2646484375
          # 0.23732948 0.2496337890625
          exit
```

**Comment by sagadre** [comment]
Hi! Looking over your code the two major differences I noticed between your implementation and the one found in `https://github.com/mlfoundations/dataset2metadata` are
1. we use the openai CLIP repo directly for the preprocessing, though the weights are the same and this should not be an issue
2. we conduct inference in half precision (`fp16`), not `fp32` for inference

Can you try 2. and let me know if you are still running into issues? Thanks!

**Comment by mingtan2** [comment]
@sagadre , Thank you so much for the quick reply.

I found using the original un-resized images reduces the difference a lot (average absolute diff from 0.007 to 0.003). Changing to `fp16` further helps but cannot eliminate the difference completely. Anyway, I think the final difference is acceptable.

Thanks.

**Comment by sagadre** [comment]
@mingtan2 great that the differences are looking lower now. just to confirm, are you using our default `img2dataset` configuration from our [download_upstream.py](https://github.com/mlfoundations/datacomp/blob/bb02268caa28cc35b3222bcc3c1b2745afc7b2b6/download_upstream.py#L119) script? If not that could be yet another source of the discrepancy. 

as for the leaderboard, submissions are not currently updated in realtime. we have been checking submissions frequently on our end 🙂

**Comment by mingtan2** [comment]
@sagadre Yes, I used the default configs with `python3 download_upstream.py --scale medium --data_dir /path/to/medium --processes_count 64`.

Thanks for the info about the leaderboard update.

**Comment by mingtan2** [comment]
@sagadre Can I know why a few datasets have disabled the `cache_dir`, e.g., [here](https://github.com/mlfoundations/datacomp/blob/main/eval_utils/retr_eval.py#L41)? In addition, it seems `flick_1k` cannot be downloaded using `download_evalsets.py` due to "Username/Password Authentication Failed." I need this because my some of my compute notes have no network access and need offline eval. Thanks a lot.

**Comment by sagadre** [comment]
tagging @djghosh13, who will know `cache_dir` and `flick_1k` specifics

**Comment by djghosh13** [comment]
@mingtan2 sorry for the inconvenience! Some of the datasets were added later as Huggingface datasets (as opposed to webdataset for the original datasets), and so were not included in the `download_evalsets.py` script. I'll push an update to fix this soon. Additionally, the `flickr_1k` error is most likely because it's trying to download a webdataset that doesn't exist.

**Comment by djghosh13** [comment]
Submitted a PR to fix this at #21; once it goes through you should be able to re-run `download_evalsets.py` without error, and run `evaluate.py` as intended (specifying `--data_dir`)

**Comment by mingtan2** [comment]
@sagadre Can I know when the leaderboard will be updated and the updating frequency? Thanks!
